packages/taquito-michel-codec/src/errors.ts

```diff
@@ -1,78 +1,79 @@
+import { ParameterValidationError, TaquitoError } from '@taquito/core';
 /**
  *  @category Error
- *  @description Error that indicates an invalid contract being passed or used
+ *  @description Error indicates an invalid Michelson being passed or used
  */
-export class InvalidContractError extends Error {
-  public name = 'InvalidContractError';
+export class InvalidMichelsonError extends ParameterValidationError {
   constructor(public readonly message: string) {
     super();
+    this.name = 'InvalidMichelsonError';
   }
 }
 
 /**
  *  @category Error
- *  @description Error that indicates an invalid type expression being passed or used
+ *  @description Error indicates an invalid type expression being passed or used
```

Comment:
this is a nitpick, but I think using 'that' makes a bit more sense in documentation. Using passive voice in a doc is a bit easier to understand imo

Reply:
updated all error class @description typedoc as suggestion to be consistent